Updates BSK to 67.0.0

diegoreymendez opened the pull request:
Task/Issue URL: https://app.asana.com/0/0/1205009979720480/f

BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/414
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1345

## Description:

This PR changes how the System Network Extension notifies "DuckDuckGo Notifications" that it needs to show a notification. We're now using Distributed Notifications for this goal.

**Steps to test this PR**:

Just make sure the iOS App builds well.  None of the changes should affect the iOS App.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

diegoreymendez committed:
Updates BSK

graeme reviewed (approved):
Looks good. Just point to the BSK version and also update the PR name to specify the BSK version too, then it’s good to go.

diegoreymendez committed:
Updates BSK due to a macOS changes (#1826)

Task/Issue URL: https://app.asana.com/0/inbox/1203108348814444/1205013436492168/1205012974984921

BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/415
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1346

## Description:

Updates BSK to bring over some changes for macOS.

**Steps to test this PR**:

Make sure the iOS app builds.  These changes should not affect iOS in any way.

diegoreymendez committed:
Merges the latest from develop and updates to BSK 67.0.0

diegoreymendez committed:
Updates to BSK 67.0.0
